src/AbstractMCMC.jl

```diff
@@ -88,4 +88,19 @@ include("stepper.jl")
 include("transducer.jl")
 include("logdensityproblems.jl")
 
+function __init__()
+    if isdefined(Base.Experimental, :register_error_hint)
+        Base.Experimental.register_error_hint(MethodError) do io, exc, argtypes, _
+            if Base.parentmodule(exc.f) == LogDensityProblems &&
+                any(a -> a <: LogDensityModel, argtypes)
+                printstyled(
+                    io,
+                    "\nAbstractMCMC.LogDensityModel is a wrapper and does not itself implement the LogDensityProblems.jl interface. To use LogDensityProblems.jl methods, access the inner type with (e.g.) `logdensity(model.logdensity, params)` instead of `logdensity(model, params)`.";
+                    color=:cyan,
```

Comment:
I think it's safer to just print a vanilla message, without setting any color - you never know what colors people use in their REPL, so some might not be able to read text if it is displayed in an unfortunate/mismatching color.

```suggestion
                print(
                    io,
                    "\n`AbstractMCMC.LogDensityModel` is a wrapper and does not itself implement the LogDensityProblems.jl interface. To use LogDensityProblems.jl methods, access the inner type with (e.g.) `logdensity(model.logdensity, params)` instead of `logdensity(model, params)`.",
```